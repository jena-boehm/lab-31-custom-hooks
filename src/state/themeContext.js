
import React, { createContext, useContext, useEffect, useState } from 'react';

export const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('dark');

  const toggleTheme = (currentTheme) => {
    if(currentTheme === 'light') setTheme('light');
    else setTheme('dark');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const { theme, toggleTheme } = useContext(ThemeContext);

  return {
    theme,
    toggleTheme
  };
};

