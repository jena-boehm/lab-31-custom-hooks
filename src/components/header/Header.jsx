import React from 'react';
import { Link } from 'react-router-dom';
import styles from '../app/styles/App.css';
import logo from '../../../public/pawprint.png';
import { useTheme } from '../../state/themeContext';

const Header = () => {
  const { theme, toggleTheme } = useTheme();

  const changeTheme = ({ target }) => {
    if(target.checked) toggleTheme('light');
    else toggleTheme('dark');
  };

  return (
    <div className={styles.header}>
      <img 
        src={logo}
        alt="logo" 
        width="70px" 
        className={styles.logo} />
      <Link 
        to="/"
        className={styles.headerLink}>Home</Link> 
      <div className={styles.toggleContainer}>
        <span>Check to toggle light/dark mode:</span>
        <input 
          className={styles.headerLink}
          type="checkbox" 
          onChange={changeTheme} />
      </div>
    </div>
  );
};

export default Header;
