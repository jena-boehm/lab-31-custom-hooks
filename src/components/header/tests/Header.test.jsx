import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from '../Header';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '../../../state/themeContext';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders Header', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </ThemeProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
