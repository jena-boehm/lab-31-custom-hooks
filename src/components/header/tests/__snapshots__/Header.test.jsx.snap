// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <img
      alt="logo"
      class="logo"
      src="../../../pawprint.png"
      width="70px"
    />
    <a
      class="headerLink"
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
